src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -203,6 +214,39 @@ public void visitColumnWriteFragment(ColumnWriteFragment columnWriteFragment) {
     }
 
     // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    // Table Mutation: deletion
+
+    @Override
+    public void visitStandardTableDelete(TableDeleteStandard tableDeleteStandard) {
+        if (tableDeleteStandard.getWhereFragment() != null) {
+            // only applies when the table is a collection table with where restriction
+            throw new FeatureNotSupportedException();
+        }
+
+        if (tableDeleteStandard.getNumberOfKeyBindings() > 1) {
+            throw new FeatureNotSupportedException("MongoDB doesn't support id spanning across multiple columns");
+        }
+        assertTrue(tableDeleteStandard.getNumberOfKeyBindings() == 1);
+
+        var tableName = tableDeleteStandard.getTableName();
+
+        var filters = new ArrayList<AstFilter>(1 + tableDeleteStandard.getNumberOfOptimisticLockBindings());
+        filters.add(getAstFieldFilter(tableDeleteStandard.getKeyBindings().get(0)));
+        tableDeleteStandard.forEachOptimisticLockBinding(
+                (idx, optimisticLockBinding) -> filters.add(getAstFieldFilter(optimisticLockBinding)));
+
+        astVisitorValueHolder.yield(COLLECTION_MUTATION, new AstDeleteCommand(tableName, new AstAndFilter(filters)));
```

Comment:
We need to accommodate optimistic locking fields; otherwise the important optimisitc locking mechanism (https://docs.jboss.org/hibernate/orm/6.6/userguide/html_single/Hibernate_User_Guide.html#locking-optimistic) won't work.